src/Authoring/WinRT.SourceGenerator2/TypeMapAssemblyTargetGenerator.Execute.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+#pragma warning disable IDE0008
+
+namespace WindowsRuntime.SourceGenerator;
+
+/// <inheritdoc cref="TypeMapAssemblyTargetGenerator"/>
+public partial class TypeMapAssemblyTargetGenerator : IIncrementalGenerator
+{
+    /// <summary>
+    /// Generation methods for <see cref="TypeMapAssemblyTargetGenerator"/>.
+    /// </summary>
+    private static class Execute
+    {
+        /// <summary>
+        /// Gets all portable executable references from a compilation.
+        /// </summary>
+        /// <param name="compilation">The input compilation.</param>
+        /// <param name="token">The cancellation token for the operation.</param>
+        /// <returns>All portable executable references for <paramref name="compilation"/>.</returns>
+        public static ImmutableArray<EquatablePortableExecutableReference> GetAllPortableExecutableReferences(Compilation compilation, CancellationToken token)
+        {
+            var executableReferences = ImmutableArray.CreateBuilder<EquatablePortableExecutableReference>();
+
+            foreach (MetadataReference metadataReference in compilation.References)
+            {
+                token.ThrowIfCancellationRequested();
+
+                // We are only interested in PE references (not project references)
```

Comment:
When running on the command line, there are no project references, only PE references.